c7n/resources/datapipeline.py

```diff
@@ -187,19 +168,13 @@ class TagPipeline(Tag):
 
     permissions = ('datapipeline:AddTags',)
 
-    def process_resource_set(self, pipelines, tags):
-        client = local_session(self.manager.session_factory).client(
-            'datapipeline')
-        tag_array = []
-        for t in tags:
-            tag_array.append(dict(key=t['Key'], value=t['Value']))
+    def process_resource_set(self, client, pipelines, tags):
+        tag_array = [dict(key=t['Key'], value=t['Value']) for t in tags]
```

Comment:
TIL datapipeline uses lowercase 'key' and 'value' for the add_tags call https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/datapipeline.html#DataPipeline.Client.add_tags